OctopusDSC/DSCResources/cOctopusServer/cOctopusServer.psm1

```diff
@@ -359,12 +358,17 @@ function Set-TargetResource {
         Stop-OctopusDeployService $Name
     }
 
-    if ($Ensure -eq "Absent" -and $currentResource["Ensure"] -eq "Present") {
-        Uninstall-OctopusDeploy $Name
+    if ($Ensure -eq "Absent" -and (($currentResource["Ensure"] -eq "Present") -or ($currentResource["State"] -eq "Installed"))) {
```

Comment:
Might just be monday brain but these if/else statements are getting hard to read